components/brave_rewards/browser/publisher_info_database.cc

```diff
@@ -1044,29 +1044,100 @@ bool PublisherInfoDatabase::MigrateV1toV2() {
     return false;
   }
 
-  if (!CreateContributionInfoTable()) {
+  const char* name = "contribution_info";
+  sql = "CREATE TABLE ";
+  sql.append(name);
+  sql.append(
+      "("
+      "publisher_id LONGVARCHAR,"
+      "probi TEXT \"0\"  NOT NULL,"
+      "date INTEGER NOT NULL,"
+      "category INTEGER NOT NULL,"
+      "month INTEGER NOT NULL,"
+      "year INTEGER NOT NULL,"
+      "CONSTRAINT fk_contribution_info_publisher_id"
+      "    FOREIGN KEY (publisher_id)"
+      "    REFERENCES publisher_info (publisher_id)"
+      "    ON DELETE CASCADE)");
+  if (!GetDB().Execute(sql.c_str())) {
+    return false;
+  }
+
+  if (!GetDB().Execute(
+      "CREATE INDEX IF NOT EXISTS contribution_info_publisher_id_index "
+      "ON contribution_info (publisher_id)")) {
     return false;
   }
 
-  if (!CreateContributionInfoIndex()) {
+  // Recurring_donation
```

Comment:
Why is table dropped here?

Reply:
This was from v2 creation script. I'm assuming table didn't exist in v1?

Reply:
This table didn't exist in v1 and is dropped if exists so that it can be created correctly: https://github.com/brave/brave-core/commit/3271f430a786f537e7e3a0b5cbeb5c0725e35f9b#diff-4bf8d740c793a0cf750bc637b07f65a0R604